client/jetpack-cloud/sections/partner-portal/invoices-list/index.tsx

```diff
@@ -38,8 +40,20 @@ const InvoicePlaceholderCard = memo( () => {
 
 export default function InvoicesList() {
 	const translate = useTranslate();
+	const dispatch = useDispatch();
 	const [ pagination, setPagination ] = useState( { starting_after: '', ending_before: '' } );
 	const invoices = useInvoicesQuery( pagination );
+
+	useEffect( () => {
+		if ( invoices.isError ) {
+			dispatch(
+				errorNotice( translate( 'We were unable to retrieve your invoices.' ), {
+					id: 'partner-portal-invoices-failure',
```

Comment:
Dispatching the `errorNotice` doesn't come with any cleanup by default, so should we add some expiration? Otherwise, the notice will just stay there forever, including if `invoices` end up loaded properly on the next request. Alternatively, we could do a `removeNotice()` by that same ID if `! invoices.isError`.

Reply:
I added a notice duration of 5s 👍🏻 I also experimented with automatically cleaning up the notice if the query is successful but I have some [thoughts](https://tkdodo.eu/blog/react-query-error-handling#putting-it-all-together) on improving errors for queries in general so we can don't have to rely on `useEffect` at all but that's for another PR.